keanu-project/src/test/java/io/improbable/keanu/algorithms/tensormcmc/TensorNUTSTest.java

```diff
@@ -0,0 +1,80 @@
+package io.improbable.keanu.algorithms.tensormcmc;
+
+import io.improbable.keanu.algorithms.NetworkSamples;
+import io.improbable.keanu.algorithms.mcmc.MCMCTestDistributions;
+import io.improbable.keanu.algorithms.mcmc.NUTS;
+import io.improbable.keanu.algorithms.mcmc.tensor.TensorNUTS;
+import io.improbable.keanu.network.BayesNetDoubleAsContinuous;
+import io.improbable.keanu.network.BayesNetTensorAsContinuous;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.dbltensor.DoubleTensor;
+import io.improbable.keanu.vertices.dbltensor.KeanuRandom;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TensorNUTSTest {
+
+    private KeanuRandom random;
+
+    @Before
+    public void setup() {
+        random = new KeanuRandom(1);
+    }
+
+    @Test
+    public void samplesGaussian() {
```

Comment:
We should add a test with a multivariate normal when we add it as a distribution. I'll add that as a part of the ACs for the multivariate normal task.

Reply:
Nice man, let's make sure we add it to `Hamiltonian` to